### ISSUE TITLE
As a Project coordinator, I want a confirmation when I discard a project revision, so that I don't do it by mistake

### ISSUE THREAD

**Issue by matthieu-foucault** [bcgov/cas-cif#234]
#### Description:

A confirmation step is required before discarding a project revision, since this deletes all the related `form_change` records including entered-but-not-yet-committed data.
Accidentally discarding a project revision could result in unwanted data loss.

#### Acceptance Criteria:

Given I have begun a revision of an existing project
When I want to scrap all changes I have made in this revision
Then there is an option to do so

Given I am editing a project revision
When I select the option to discard all changes
Then I can see a confirmation that all the changes that will be discarded

Given I am editing a project revision
And I clicked the discard revision button
When I confirm that I want to discard the revision and all related changes
Then that revision is discarded, deleted from the database and will no longer show up in the UI

Given I am discarding a project revision
And I confirmed that I wanted to delete the revision 
Then I am taken to
  - either the projects list if this was a "create project" revision
  - or the project view page if this was an "edit project" revision

#### Development Checklist:

- [x] A discard revision button exists in the edit project UI
- [x] The discard revision button should only appear on the review page below 'Review and Submit Project
- [x] Clicking the discard revision button shows the confirmation message in an alert along with a confirm / cancel button
- [x] Confirming the discard, deletes the revision and all the related form changes from the database
- [ ] Meets the DOD

#### Notes:

-

**Comment by dleard** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Climate-Action-Secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/234?planning-poker) with ZenHub @AlexZorkin @pbastia

**Comment by pbastia** [comment]
![image.png](https://images.zenhubusercontent.com/5faafee70e671ac198e88bc7/ae168a9b-4499-4f06-80ea-16941eae64c2)

**Comment by pbastia** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Climate-Action-Secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/234?planning-poker) with ZenHub @BCerki @gurjmatharu @matthieu-foucault @mikevespi @Sepehr-Sobhani

**Comment by suhafa** [comment]
![image.png](https://images.zenhubusercontent.com/6201b10c9827341d8d5cbe9c/38cb03bb-6ee3-41f7-8d46-735a15738a24)
Updated wireframe (without 'Undo Changes' Button)

**Comment by LindsayMacfarlane** [comment]
Thank you @dleard ! 

Tested in dev by:
- Editing an existing project: changed funding year/project name. On review and submit page, I was able to select the discard project revision button, I received the error message and selected proceed and redirected back to the project overview page.
- Creating a new project: I was able to enter new project information, go to review and submit page, select the discard project revision button, received the error message and get redirected back to project lists. 

Observation
- When creating a new project, should the button be called 'project revision'? As a user, I'd be entering new information, not revising existing information. Just a thought.
